### PR TITLE
Add Tiny Pumpkin.

### PR DESCRIPTION
**Pull request by vini2003**
I believe all citizens of Minecraft should have access to their own Tiny Pumpkin. As such, I have added it.

### CODE REVIEW

File: src/main/java/com/fabriccommunity/spookytime/common/SpookyItems.java [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
     private static Item register(String name, Item item)
     {
         return Registry.register(Registry.ITEM, new Identifier(SpookyTime.MODID, name), item);
     }
 
+    private static Item register(String name, BlockItem item)
+    {
+        return Registry.register(Registry.ITEM, new Identifier(SpookyTime.MODID, name), item);
+    }
+
```
Comment (Shrimpstress) [inline src/main/java/com/fabriccommunity/spookytime/common/SpookyItems.java]: I'm not sure why a new method is necessary? A BlockItem is still an Item so the existing register method should work just fine.

File: src/main/java/com/fabriccommunity/spookytime/common/block/TinyPumpkinBlock.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public VoxelShape getCollisionShape(BlockState blockState, BlockView blockView, BlockPos blockPosition, EntityContext entityContext) {
+    	return this.collidable ? blockState.getOutlineShape(blockView, blockPosition) : VoxelShapes.empty();
+    }
+
+    @Override
+    public BlockRenderLayer getRenderLayer() {
+    	return BlockRenderLayer.CUTOUT;
+    }
```
Comment (Shrimpstress) [inline src/main/java/com/fabriccommunity/spookytime/common/block/TinyPumpkinBlock.java]: I don't think the CUTOUT layer is necessary in this case.

File: src/main/java/com/fabriccommunity/spookytime/common/SpookyItems.java [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
 
 public class SpookyItems
 {
 
 
+    public static BlockItem TINY_PUMPKIN;
+
     public static void init() {
-        // NO-OP
+        register("tiny_pumpkin", TINY_PUMPKIN = new BlockItem(SpookyBlocks.TINY_PUMPKIN, new Item.Settings().group(ItemGroup.MISC)));
```
Comment (Shrimpstress) [inline src/main/java/com/fabriccommunity/spookytime/common/SpookyItems.java]: The same as when you registered the Block in SpookyBlocks below.

File: src/main/java/com/fabriccommunity/spookytime/common/SpookyBlocks.java [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
 import net.minecraft.util.registry.Registry;
 
 public class SpookyBlocks
 {
 
 
+    public static Block TINY_PUMPKIN;
+
     public static void init() {
+        register("tiny_pumpkin", TINY_PUMPKIN = new TinyPumpkinBlock(FabricBlockSettings.of(Material.PUMPKIN).strength(1.0F, 1.0F).sounds(BlockSoundGroup.LANTERN).build()));
```
Comment (Shrimpstress) [inline src/main/java/com/fabriccommunity/spookytime/common/SpookyBlocks.java]: This is more of a style thing, but the register method returns a block, so I would change this to 
`TINY_PUMPKIN = register("tiny_pumpkin", new TinyPumpkinBlock(FabricBlockSettings.of(Material.PUMPKIN).strength(1.0F, 1.0F).sounds(BlockSoundGroup.LANTERN).build()));`

**Review by Shrimpstress — CHANGES_REQUESTED** [review]

File: src/main/java/com/fabriccommunity/spookytime/common/block/TinyPumpkinBlock.java (hunk 2 above)
Comment (vini2003) [reply src/main/java/com/fabriccommunity/spookytime/common/block/TinyPumpkinBlock.java]: Unsure if `CUTOUT` is, though it does need to be overriden.

Comment (Shrimpstress) [reply src/main/java/com/fabriccommunity/spookytime/common/block/TinyPumpkinBlock.java]: Why does it need to be overriden? You don't seem to be using any textures with transparency currently. The normal render layer should be perfectly acceptable in this case.

**Comment by vini2003** [comment]
This last commit:
- Reverts the cursed Gradle PR #4;
- Adds a way to make Tiny Pumpkin;
- Adds drops to Tiny Pumpkin;
- Adds Mixin to do such;
- Fixes inconsistency with JSON files - tabs, spaces, four spaces, two spaces, it wasn't beautiful.
- Fixes inconsistency with Java files - brackets opening seemingly randomly, unless there's a code style I'm missing there.

**Review by liach — APPROVED** [review]

**Comment by vini2003** [comment]
This PR should be good for reviewing now.

File: src/main/java/com/fabriccommunity/spookytime/common/SpookyBlocks.java [hunk 5]
```diff
@@ -1,21 +1,28 @@
 package com.fabriccommunity.spookytime.common;
 
 import com.fabriccommunity.spookytime.SpookyTime;
+import com.fabriccommunity.spookytime.block.TinyPumpkinBlock;
+
+import net.fabricmc.fabric.api.block.FabricBlockSettings;
 import net.minecraft.block.Block;
+import net.minecraft.block.Material;
+import net.minecraft.entity.passive.CowEntity;
```
Comment (Shnupbups) [inline src/main/java/com/fabriccommunity/spookytime/common/SpookyBlocks.java:9]: Why is CowEntity imported into SpookyBlocks?

File: src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    protected static final VoxelShape Y_SHAPE = Block.createCuboidShape(6.0D, 0.0D, 6.0D, 10.0D, 8.0D, 10.0D);    
+
+    public TinyPumpkinBlock(Settings blockSettings) {
+        super(blockSettings);
+    }
+
+    @Override
+    public BlockState getPlacementState(ItemPlacementContext placementContext) {
+        return (BlockState)this.getDefaultState().with(FACING, placementContext.getPlayerFacing().getOpposite());
```
Comment (Shnupbups) [inline src/main/java/com/fabriccommunity/spookytime/block/TinyPumpkinBlock.java]: Unnecessary cast to BlockState

File: src/main/resources/assets/spookytime/lang/en_us.json [hunk 7]
```diff
@@ -1,3 +1,5 @@
 {
     "block.spookytime.tiny_pumpkin": "§4T§6i§en§ay §4P§6u§em§ap§9k§di§6n"
```
Comment (liach) [inline src/main/resources/assets/spookytime/lang/en_us.json]: need a comma here

**Review by Shnupbups — APPROVED** [review]
Looks good, some inconsequential nitpicks, but otherwise good to merge

File: src/main/resources/assets/spookytime/models/item/tiny_pumpkin.json [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	"display": {
+        "gui": {
+            "rotation": [ 30, 225, 0 ],
+            "translation": [ 0, 0, 0],
+            "scale":[ 0.625, 0.625, 0.625 ]
+        },
+        "ground": {
+            "rotation": [ 0, 0, 0 ],
+            "translation": [ 0, 3, 0],
+            "scale":[ 0.25, 0.25, 0.25 ]
```
Comment (liach) [inline src/main/resources/assets/spookytime/models/item/tiny_pumpkin.json:12]: The spacing of json is weird here

**Review by ImAK9 — APPROVED** [review]

**Pull request merged**